lib/pengine/utils.c

```diff
@@ -215,10 +216,44 @@ node_list_dup(GListPtr list1, gboolean reset, gboolean filter)
 gint
 sort_node_uname(gconstpointer a, gconstpointer b)
 {
-    const node_t *node_a = a;
-    const node_t *node_b = b;
-
-    return strcmp(node_a->details->uname, node_b->details->uname);
+    const char *name_a = ((const node_t *) a)->details->uname;
+    const char *name_b = ((const node_t *) b)->details->uname;
+
+    while (*name_a && *name_b) {
+        if (isdigit(*name_a) && isdigit(*name_b)) {
+            // If node names contain a number, sort numerically
+            char *end_a = NULL;
+            char *end_b = NULL;
+            long num_a = strtol(name_a, &end_a, 10);
+            long num_b = strtol(name_b, &end_b, 10);
+
+            if (num_a < num_b) {
+                return -1;
+            } else if (num_a > num_b) {
+                return 1;
+            }
+            name_a = end_a;
+            name_b = end_b;
```

Comment:
This won't discern `node0` from `node00`, which may explode somewhere
up the call stack (e.g. checking that no duplicates are present if it turns up
important in the future).  Better to mix comparison on the respective numbers
of consumed characters in.

Reply:
Good catch, thanks :)